usr/lib/xtable/luaxt.c

```diff
@@ -0,0 +1,239 @@
+/*
+* SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <lauxlib.h>
+#include <lua.h>
+#include <lualib.h>
+
+#include <errno.h>
+#include <xtables.h>
+
+#include "luaxtable.h"
+
+#ifndef LUAXTABLE_MODULE
+#error "LUAXTABLE_MODULE is not defined"
+#endif
+
+#define pr_err(fmt, ...) fprintf(stderr, fmt, ##__VA_ARGS__)
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+typedef struct luaxt_flags_s {
+	const char *name;
+	lua_Integer value;
+} luaxt_flags_t;
+
+static lua_State *L = NULL;
+static int luaopen_luaxt(lua_State *L);
+static int luaxt_match(lua_State *L);
+static int luaxt_target(lua_State *L);
+
+static int luaxt_run(lua_State *L, const char *func_name, const char *key, int nargs, int nresults)
+{
+	int ret = -1;
+	int base = lua_gettop(L) - nargs;
+	lua_rawgetp(L, LUA_REGISTRYINDEX, key);
+
+	if (lua_getfield(L, -1, func_name) != LUA_TFUNCTION) {
+		pr_err("Function %s not found\n", func_name);
+		goto restore;
+	}
+
+	lua_insert(L, base + 1); /* func */
+	lua_pop(L, 1); /* table */
+
+	if (lua_pcall(L, nargs, nresults, 0) != LUA_OK) {
+		pr_err("failed to call Lua function %s: %s\n", func_name,
+				lua_tostring(L, -1));
+		goto restore;
+	}
+
+	if (nresults == 1)
+		ret = lua_toboolean(L, -1);
+restore:
+	lua_settop(L, base);
+	return ret;
+}
+
+static int luaxt_runwithuserargs(lua_State *L, const char *op, const char *key, unsigned int *flags, luaxtable_info_t *info)
+{
+	int ret;
+
+	lua_newtable(L);
+	lua_pushvalue(L, -1); /* stack : param, param */
+
+	if((ret = luaxt_run(L, op, key, 1, 1)) == -1)
+		return 0;
+
+	if (flags && (lua_getfield(L, -1, "flags") == LUA_TNUMBER)) {
+		*flags = lua_tointeger(L, -1);
+		lua_pop(L, 1);
+	}
+
+	if (lua_getfield(L, -1, "userargs") == LUA_TSTRING) {
+		size_t len = 0;
+		const char *ldata = lua_tolstring(L, -1, &len);
+		memset(info->userargs, 0, LUAXTABLE_USERDATA_SIZE);
```

Comment:
you can just `memset` the rest (that's it, `LUAXTABLE_USERDATA_SIZE - MIN(len, LUAXTABLE_USERDATA_SIZE)`), right? after copying ldata..

Reply:
It will still result in a memset call, right ?

Reply:
yeah, but it would memset only the reminder of the "buffer".. perhaps it's a bit of premature optimization from my side =).. I just found it a bit weird to memset to 0 and immediately after memcpy to the very same portion of the memory.. but we can leave it as is.. it's up to you =)

Reply:
We can optimise it later if needed.